src/routes/safe/components/Apps/ManageApps.js

```diff
@@ -79,15 +75,23 @@ const ManageApps = ({ appList, onAppAdded, onAppToggle }: Props) => {
 
   const cleanAppInfo = () => setAppInfo(APP_INFO)
 
-  const safeAppValidator = async (value) => {
-    const appInfo = await getAppInfoFromUrl(value)
-
-    if (appInfo.error) {
+  const customRequiredValidator = (value) => {
+    if (!value || !value.length) {
       setAppInfo(APP_INFO)
-      return 'This is not a valid Safe app.'
+      return 'Required'
     }
+  }
 
-    setAppInfo({ ...appInfo })
+  const getIpfsLinkFromEns = async (name) => {
+    try {
+      const content = await getContentFromENS(name)
+      if (content && content.protocolType === 'ipfs') {
+        return `https://ipfs.io/ipfs/${content.decoded}/`
+      }
+    } catch (error) {
+      console.error(error)
+      return undefined
+    }
   }
 
   const uniqueAppValidator = (value) => {
```

Comment:
This doesn't have to be defined inside a component

Reply:
yes it does, it makes use of `appList`

Reply:
you can refactor this validator to use currying
check `uniqueAddress` validator